DataCollection/src/Services/Controllers/WaterbirdForagingSurveyController.cs

```diff
@@ -31,6 +38,72 @@ public WaterbirdForagingSurveyController(IWaterbirdForagingManager domainManager
         }
 
         /// <summary>
+        /// Retrieves the requested Waterbird Foraging Survey data.
+        /// </summary>
+        /// <param name="surveyIdentifier">
+        /// Unique identifier for the survey resource to retrieve.
+        /// </param>
+        /// <returns>
+        /// <see cref="IHttpActionResult"/> containiing a <see cref="WaterbirdForagingModel"/> or status 404 if none found.
+        /// </returns>
+        [HttpGet]
+        [Authorize]
+        public IHttpActionResult Get(Guid surveyIdentifier)
+        {
+            return WrapWithTryCatch(() =>
+            {
+                var result = _domainManager.FindBySurveyId(surveyIdentifier);
+
+                if (result == null)
+                {
+                    return NotFound();
+                }
+
+                var model = Map(result);
+
+                return Ok(model);
+            });
+        }
+
+        /// <summary>
+        /// Retrieves a a list of Waterbird Foraging information for a given user / submitter.
+        /// </summary>
+        /// <param name="userId">
+        /// UserId of the person who submitted the survey.
+        /// </param>
+        /// <returns>
+        /// <see cref="IHttpActionResult"/> containing a list of <see cref="WaterbirdForagingListItem"/> or 404 if none found.
+        /// </returns>
+        [HttpGet]
+        [Authorize]
+        public IHttpActionResult Get(int userId)
+        {
+            return WrapWithTryCatch(() =>
+            {
+                var result = _domainManager.FindBySubmitterId(userId);
+
+                if (result == null || !result.Any())
+                {
+                    return NotFound();
+                }
+
+                var models = result.Select(x => {
+                    return new WaterbirdForagingListItem
```

Comment:
I should extract this mapping to a stand-alone function, so that `Get` is not doing "more than one thing" (its "one thing" is to handle Controller I/O, not to map data between data types).